usrsctplib/netinet/sctputil.c

```diff
@@ -6716,7 +6716,7 @@ sctp_sorecvmsg(struct socket *so,
 						sctp_sblog(&so->so_rcv, control->do_not_ref_stcb?NULL:stcb,
 							   SCTP_LOG_SBRESULT, 0);
 					}
-					atomic_subtract_int(&control->length, cp_len);
+					atomic_subtract_int(&control->length, (int32_t)cp_len);
```

Comment:
Fixed in https://github.com/sctplab/usrsctp/commit/078bd24ac1768bc6dc87d40b030434cba1987ea3.